File: 조준장/level2/롤케이크 자르기.py
from collections import Counter

def solution(topping):
    answer = 0
    a = Counter(topping)
    b = set()
    
    for t in topping:
        a[t] -= 1
        b.add(t)
        
        if a[t] == 0:
            del a[t]
    
        if len(a) == len(b):
            answer += 1
    
    return answer

File: 조준장/level2/롤 케이크 자르기.kt
class Solution {
    fun solution(topping: IntArray): Int {
        var answer: Int = 0

        val a = mutableMapOf<Int, Int>()
        val b = mutableSetOf<Int>()

        topping.toSet().forEach { t ->
            a += t to topping.count { it == t }
        }
        
        topping.forEach {
            a[it] = a[it]!! - 1
            b.add(it)
            
            if (a[it] == 0) a.remove(it)
            if (a.size == b.size) answer ++
        }

        return answer
    }
}
